connector/src/main/java/org/jboss/as/connector/services/driver/registry/DriverRegistryImpl.java

```diff
@@ -84,19 +85,33 @@ public void unregisterInstalledDriver(InstalledDriver driver) {
 
     @Override
     public Set<InstalledDriver> getInstalledDrivers(String profileName) {
-        String profile = profileName != null ? profileName : DEFAULT_PROFILE;
-        if (drivers.get(profile) == null) {
-            throw ROOT_LOGGER.noDriverDefinedInDefaultProfile();
+        Set installedDrivers = new HashSet<InstalledDriver>();
+        if (profileName != null) {
+            if (drivers.get(profileName) == null) {
+                throw ROOT_LOGGER.noDriverDefinedInDefaultProfile();
+            }
+
+            installedDrivers.addAll(drivers.get(profileName).values());
+        }
+        if (drivers.get(DEFAULT_PROFILE) != null) {
+            installedDrivers.addAll(drivers.get(DEFAULT_PROFILE).values());
         }
-        return Collections.unmodifiableSet(Collections.synchronizedSet(new HashSet<InstalledDriver>(drivers.get(profile).values())));
+
+        return Collections.unmodifiableSet(Collections.synchronizedSet(installedDrivers));
```

Comment:
Collections.synchronizedSet shouldn't be needed, since installedDrivers is now a copy. OTOH, there's a pre-existing problem where the whole method should be in a synchronized (drivers) block as that field is not thread safe.